### DISCUSSION TITLE
WYWISYG Editor with code highlighting feature.

### DISCUSSION THREAD

**Discussion by mohsinamjad** [directus/directus#4198] [Feature Requests]
It's feature request for implementing highlight.js or some other module in WYWISYG editor for code highlighting 
right now the code-block doesn't highlight code snippets and it doesn't look good overall if you are writing some technical blog.
code samples are required in technical blogs just like blogs/articles on dev.to or markdown editors, it will be a great feature for developers :D 

![Screenshot from 2021-02-20 22-30-12](https://user-images.githubusercontent.com/32832044/108604142-cffebc00-73cd-11eb-8367-24d4f13a23bb.png)

**Comment by philleepflorence** [comment]
The D9 WYSIWYG Interface is powered by TinyMCE v5, which should already come with the **Code Sample plugin** and the ability to _Insert and embed syntax highlighted code snippets_.

> By default, codesample uses http://prismjs.com/ to embed the code samples within the editor and works out of the box.

[Code Sample plugin - Tiny Docs](https://www.tiny.cloud/docs/plugins/opensource/codesample/)

> You need to add prism.js and prism.css to your page in order to get the syntax highlighted code samples on your webpage (as created by the Code Sample plugin).

Is this the same as the feature request?

if so here are the ways you can extend or manage the WYSIWYG Interface:
| Option              | Description                                                                                      | Default      |
|---------------------|--------------------------------------------------------------------------------------------------|--------------|
| `toolbar`           | What toolbar options to show                                                                     | _See below_  |
| `custom-formats`    | What custom html blocks to show in the editor                                                    | `null`       |
| `font`              | Font to render the value in (`sans-serif`, `serif`, or `monospace`)                              | `serif` |
| `tinymce-overrides` | Override any of the [init options](https://www.tiny.cloud/docs/configure/integration-and-setup/) | `null` |

**Reply by mohsinamjad** [reply]
Thanks bro, I was able to configure codesample plugin in TinyMCE, however it was really difficult to add prism.css, tried all methods in their docs, in the end it started working by adding style link for https://unpkg.com/prismjs/themes/prism-tomorrow.css in tinymce content_css prop. 

![Screenshot from 2021-02-22 23-30-55](https://user-images.githubusercontent.com/32832044/108753185-39193780-7566-11eb-9714-4802b6186664.png)

**Reply by philleepflorence** [reply]
Ah! Good to know...Yes, [Content CSS](https://www.tiny.cloud/docs/configure/content-appearance/#content_css) is also how I import stylesheets.
So the **WYSIWYG Interface** and the **Website** both show the Prism syntax?

**Reply by mohsinamjad** [reply]
Did you mean **Frontend Website** ? if so then I haven't started working on it yet, just started exploring **Directus** but I guess it will work by just adding prism.js and prism.css. 
In Directus it's working fine in **WYSIWYG Interface**.

**Reply by philleepflorence** [reply]
Yes, the Frontend, okay cool.
This may come in handy for others trying to add syntax highlighting into the WYSIWYG Interface.

**Reply by mohsinamjad** [reply]
yeah, will update here once I start working on the frontend,

**Reply by sarkaramitabh300** [reply]
Hi can you please explain the step by step process, how you added the code blocks in the WYSIWYG interface

**Comment by iksent** [comment]
Hello!
Can you please provide more details on how to configure code blocks support for Directus Wysiwygs?
For now I can't just see this options for toolbar...
And I can't understand, how to enable prismjs for Directus.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by sarkaramitabh300** [reply]
Please reopen this issue, the code blocks are a must-have in any WYSIWYG editor.

**Reply by rijkvanzanten** [reply]
I'd recommend using the Markdown wysiwyg instead of the html wywisyg then 👍🏻 

Also I won't reopen this. There've only been 3 upvotes, so even though you personally feel like it's a must-have, unfortunately the community at large doesn't seem to share that opinion
